cpp/src/io/parquet/decode_fixed.cu

```diff
@@ -1167,8 +1167,9 @@ CUDF_KERNEL void __launch_bounds__(decode_block_size_t, 8)
     // For large strings, update the initial string buffer offset to be used during large string
     // column construction. Otherwise, convert string sizes to final offsets.
     if (s->col.is_large_string_col) {
-      compute_initial_large_strings_offset(
-        s, initial_str_offsets[pages[page_idx].chunk_idx], has_lists_t);
+      auto const chunks_per_rowgroup = initial_str_offsets.size();
```

Comment:
We could be decoding pages from multiple `chunks` (row groups) in the same go, hence possible `pages[page_idx].chunk_idx % initial_str_offsets.size()`. The modulus ensures that we point to the correct `input_col_idx` for each page being decoded.

Reply:
`page.chunks` are ordered by `input col_idx` and `row_group_idx`

Reply:
I think the separate variable makes it easier to read personally especially with nested `[]`. 
I'm fine either way.

Reply:
> I think the separate variable makes it easier to read personally especially with nested [].

Makes sense. Let me resolve this.

Reply:
Did this exclusively for readability, else it may be confusing as to why we did the `%` operation. Adding a comment there as well.

Reply:
Also, trusting the compiler to optimize it out 😅